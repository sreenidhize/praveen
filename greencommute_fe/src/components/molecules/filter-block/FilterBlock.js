import { Grid, makeStyles, Typography, SvgIcon } from "@material-ui/core";
import Button from "../../atom/button/Button";

const useStyles = makeStyles((theme) => ({
  text: {
    fontSize: "20px",
    fontWeight: 600,
    fontStretch: "normal",
    fontStyle: "normal",
    lineHeight: "1.3",
    letterSpacing: "0.2px",
    textSlign: "left",
    color: theme.typography.subtitle2.color,
  },
  filterButton: {
    marginRight: "6rem",
    marginTop: "5px",
    backgroundColor: theme.palette.primary.contrastText,
  },
}));

const FilterBlock = (props) => {
  const classes = useStyles();

  return (
    <Grid container direction="row" justify="space-between">
      <Grid item direction="column" xs={11}>
        <Grid item>
          <Typography className={classes.text}>Recommended for you</Typography>
        </Grid>
        <Grid item>
          <Typography variant="h6">
            Based on your profile, skills and search history
          </Typography>
        </Grid>
      </Grid>
      <Grid contianer item xs={1} justify="flex-end">
        <Grid item>
          <Button
            variant="outlined"
            color="secondary"
            type="smallBorder"
            onClick={props.onClick}
            className={classes.filterButton}
          >
            <SvgIcon {...props}>
              <path
                fill="#000"
                d="M7 6h10l-5.01 6.3L7 6zm-2.75-.39C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39c.51-.66.04-1.61-.79-1.61H5.04c-.83 0-1.3.95-.79 1.61z"
              />
            </SvgIcon>
            Filter
          </Button>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default FilterBlock;
