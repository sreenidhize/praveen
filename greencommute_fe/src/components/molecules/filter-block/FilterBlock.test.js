import React from "react";
import FilterBlock from "./FilterBlock";
import { render } from "@testing-library/react";

describe("component renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<FilterBlock />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("block should not be empty", () => {
    const { container } = render(<FilterBlock />);

    expect(container.getElementsByTagName("div").length).toBe(6);
  });
});
