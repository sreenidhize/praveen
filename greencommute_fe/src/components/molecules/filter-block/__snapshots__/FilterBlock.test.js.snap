// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <p
          class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
        >
          Recommended for you
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Based on your profile, skills and search history
        </h6>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-filterButton-2 MuiButton-outlinedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 6h10l-5.01 6.3L7 6zm-2.75-.39C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39c.51-.66.04-1.61-.79-1.61H5.04c-.83 0-1.3.95-.79 1.61z"
                fill="#000"
              />
            </svg>
            Filter
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
